packages/stripe-integration/src/stripe-upe/stripe-ocs-payment-strategy.ts

```diff
@@ -320,84 +323,68 @@ export default class StripeOCSPaymentStrategy implements PaymentStrategy {
+        }
+    }
 
-            if (stripeError) {
-                this.stripeUPEIntegrationService.throwDisplayableStripeError(stripeError);
-                throw new PaymentMethodFailedError();
-            }
+    private async _confirmStripePaymentOrThrow(
+        methodId: string,
+        additionalActionData: StripeAdditionalActionRequired['data'],
+    ): Promise<StripeUpeResult | never> {
+        const { token, redirect_url } = additionalActionData;
+        const stripePaymentData = this.stripeUPEIntegrationService.mapStripePaymentData(
+            this.stripeElements,
+            redirect_url,
+        );
+        let stripeError: StripeError | undefined;
 
-            if (!paymentIntent) {
-                throw new RequestError();
-            }
-        } else if (
-            this.stripeUPEIntegrationService.isOnPageAdditionalAction(
-                error.body.additional_action_required,
-            )
-        ) {
-            let result;
-            let isConfirmCatchError = false;
-
-            try {
-                result = !isPaymentCompleted
-                    ? await this.stripeUPEClient.confirmPayment(stripePaymentData)
-                    : await this.stripeUPEClient.retrievePaymentIntent(token);
-            } catch (error) {
-                try {
-                    result = await this.stripeUPEClient.retrievePaymentIntent(token);
-                } catch (error) {
-                    isConfirmCatchError = true;
-                }
-            }
+        try {
+            const isPaymentCompleted = await this.stripeUPEIntegrationService.isPaymentCompleted(
+                methodId,
+                this.stripeUPEClient,
+            );
 
-            if (result?.error) {
-                this.stripeUPEIntegrationService.throwDisplayableStripeError(result.error);
+            const confirmationResult = !isPaymentCompleted
+                ? await this.stripeUPEClient?.confirmPayment(stripePaymentData)
+                : await this.stripeUPEClient?.retrievePaymentIntent(token || '');
 
-                if (this.stripeUPEIntegrationService.isCancellationError(result.error)) {
-                    throw new PaymentMethodCancelledError();
-                }
+            stripeError = confirmationResult?.error;
 
+            if (stripeError || !confirmationResult?.paymentIntent) {
                 throw new PaymentMethodFailedError();
             }
 
-            if (!result?.paymentIntent && !isConfirmCatchError) {
-                throw new RequestError();
-            }
+            return confirmationResult;
+        } catch (error: unknown) {
+            this._throwStripeError(stripeError);
 
-            const paymentPayload = this._getPaymentPayload(
-                methodId,
-                isConfirmCatchError ? token : result?.paymentIntent?.id,
-            );
+            throw new PaymentMethodFailedError();
```

Comment:
do we still need it? could we move it to the `_throwStripeError`?

Reply:
Yes, you are right, thanks.